components/AddContactsBar.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { useRouter } from "next/navigation";
+
+AddContactsBar.propTypes = {
+  name: PropTypes.string,
+  url: PropTypes.string,
+  email: PropTypes.string,
+  phone: PropTypes.string,
+};
+
+export default function AddContactsBar({ contacts, sortedContacts }) {
+  interface Contacts {
+    name: string;
+    url: string;
+    email: string;
+    phone: string;
+  }
+  const [inputName, setInputName] = useState("");
+  const [inputUrl, setInputUrl] = useState("");
+  const [inputEmail, setInputEmail] = useState("");
+  const [inputPhone, setInputPhone] = useState("");
+  const router = useRouter();
+
+  const handleInputName = (e) => {
+    setInputName(e.target.value);
+    console.log(e.target.value);
```

Comment:
would recommend removing console logs - this is just a tool to test your code - shouldn't be pushed with finished work